nixos/modules/services/web-apps/cookbook.nix

```diff
@@ -0,0 +1,116 @@
+{ config, lib, pkgs, utils, ... }:
+with lib;
+
+let
+  cfg = config.services.cookbook;
+  cookbook-config-file = pkgs.writeText "config.json" (builtins.toJSON {
+    SECRET_KEY = cfg.config.secret;
+    COOKBOOK_LOCATION = cfg.config.recipes;
+    DEFAULT_LANG = cfg.config.default-language;
+    SITE_NAME = cfg.config.site-name;
+    BASE_URL = "https://${cfg.vhost}";
+  });
+in
+{
+  options.services.cookbook = {
+    enable = mkEnableOption (mdDoc "cookbook service");
```

Comment:
remember to prepend `lib` after removing the `with lib`.

Reply:
I just noticed that I marked the previous annotation as resolved before actually pushing the changes, apologies. I have now removed the `with lib` and have explicitly inherited all the keys I need.